clippy_lints/src/inconsistent_struct_constructor.rs

```diff
@@ -71,6 +72,9 @@ impl<'tcx> LateLintPass<'tcx> for InconsistentStructConstructor {
             && let ty = cx.typeck_results().expr_ty(expr)
             && let Some(adt_def) = ty.ty_adt_def()
             && adt_def.is_struct()
+            && let Some(local_def_id) = adt_def.did().as_local()
+            && let ty_hir_id = cx.tcx.local_def_id_to_hir_id(local_def_id)
+            && !is_lint_allowed(cx, INCONSISTENT_STRUCT_CONSTRUCTOR, ty_hir_id)
```

Comment:
This can use [`fulfill_or_allow`](https://doc.rust-lang.org/nightly/nightly-rustc/clippy_utils/fn.fulfill_or_allowed.html) just before the `span_lint_and_sugg` so that `#[expect]` also works

Reply:
Sorry, I've missed this review in my notifications, Pushed a patch addressing this change request.
@rustbot review